src/NServiceBus.Core/Routing/Routers/UnicastSendRouterConnector.cs

```diff
@@ -2,127 +2,46 @@ namespace NServiceBus
     {
-        public enum RouteOption
+        public UnicastSendRouterConnector(UnicastSend.PhysicalRouter physicalRouter, UnicastSend.LogicalRouter logicalRouter)
         {
-            None,
-            ExplicitDestination,
-            RouteToThisInstance,
-            RouteToAnyInstanceOfThisEndpoint,
-            RouteToSpecificInstance
-        }
-
-        public UnicastSendRouterConnector(
-            string sharedQueue,
-            string instanceSpecificQueue,
-            string distributorAddress,
-            IUnicastSendRouter unicastSendRouter,
-            DistributionPolicy distributionPolicy,
-            Func<EndpointInstance, string> transportAddressTranslation)
-        {
-            this.sharedQueue = sharedQueue;
-            this.instanceSpecificQueue = instanceSpecificQueue;
-            this.distributorAddress = distributorAddress;
-            this.unicastSendRouter = unicastSendRouter;
-            defaultDistributionPolicy = distributionPolicy;
-            this.transportAddressTranslation = transportAddressTranslation;
+            this.physicalRouter = physicalRouter;
+            this.logicalRouter = logicalRouter;
         }
 
         public override async Task Invoke(IOutgoingSendContext context, Func<IOutgoingLogicalMessageContext, Task> stage)
         {
-            var messageType = context.Message.MessageType;
-
-            var state = context.Extensions.GetOrCreate<State>();
-
-            if (state.Option == RouteOption.RouteToThisInstance && instanceSpecificQueue == null)
-            {
-                throw new InvalidOperationException("Cannot route to a specific instance because an endpoint instance discriminator was not configured for the destination endpoint. It can be specified via EndpointConfiguration.MakeInstanceUniquelyAddressable(string discriminator).");
-            }
-
-            var sendToThisEndpointDestination = ApplyDistributorLogic(context.Extensions);
-
-            var thisEndpoint = state.Option == RouteOption.RouteToAnyInstanceOfThisEndpoint ? sendToThisEndpointDestination : null;
-            var thisInstance = state.Option == RouteOption.RouteToThisInstance ? instanceSpecificQueue : null;
-            var explicitDestination = state.Option == RouteOption.ExplicitDestination ? state.ExplicitDestination : null;
-            var destination = explicitDestination ?? thisInstance ?? thisEndpoint;
-
-            var distributionPolicy = state.Option == RouteOption.RouteToSpecificInstance ? new SpecificInstanceDistributionPolicy(state.SpecificInstance, transportAddressTranslation) : defaultDistributionPolicy;
-
-            var routingStrategy = string.IsNullOrEmpty(destination)
-                ? unicastSendRouter.Route(messageType, distributionPolicy)
-                : RouteToDestination(destination);
+            var routingStrategy = physicalRouter.Route(context);
```

Comment:
I still think that the restructuring is good this way, but I start to think we should use a different name than pyhsicalRouter here. The physical routing part is usually describing the act of resolving a transport specific transport address (instance resolving + address translation) which is also done in the logical router class.

Aren't we essentially trying to check whether we can bypass routing in the physicalRouter implementation?